lib/src/store.dart

```diff
@@ -20,12 +23,8 @@ class Store {
         try {
             checkObx(bindings.obx_opt_model(opt, model.ptr));
             if (directory != null && directory.length != 0) {
-                var cStr = new CString(directory);
-                try {
-                    checkObx(bindings.obx_opt_directory(opt, cStr.ptr));
-                } finally {
-                    cStr.free();
-                }
+                var cStr = Utf8.toUtf8(directory).cast<Uint8>();
+                checkObx(bindings.obx_opt_directory(opt, cStr));
```

Comment:
freeing necessary as well

Reply:
In the future, when we can leverage an `extension`, we could have a `Utf8.toUintPointer`.
Or, we could add that to the [ffi util](https://github.com/dart-lang/ffi) and PR.